plugins/websocket_server/WebsocketServer.cc

```diff
@@ -1047,12 +1057,20 @@ void WebsocketServer::OnAsset(int _socketId,
 
     // Construct the response message
     std::string data = BUILD_MSG(this->operations[ASSET], assetUri,
-        std::string("ignition.msgs.Bytes"), bytes.SerializeAsString());
+        bytes.GetTypeName(), bytes.SerializeAsString());
 
     // Queue the message for delivery.
     this->QueueMessage(this->connections[_socketId].get(),
         data.c_str(), data.length());
   }
+
+   ignition::msgs::StringMsg msg;
+   msg.set_data("asset_not_found");
```

Comment:
Could it be that this message is being sent regardless if the asset was found or not?

I get two `asset` messages for the same uri. One returns the bytes, and the second one with the 'asset_not_found' string.

Reply:
Fixed in [51c092d](https://github.com/gazebosim/gz-launch/pull/197/commits/51c092d5be3fc2b2d028412bfa8721dd76f56f94)